src/meoperators.jl

```diff
@@ -188,684 +178,658 @@ For the definitions of the Lyapunov operators see:
 M. Konstantinov, V. Mehrmann, P. Petkov. On properties of Sylvester and Lyapunov
 operators. Linear Algebra and its Applications 312:35–71, 2000.
 """
-function lyapop(A::AbstractMatrix{T}, E::AbstractMatrix{T}; disc = false, her = false) where T
-    GeneralizedLyapunovMap(A, E; disc = disc, her = her)
+function lyapop(A::AbstractMatrix{T1}, E::AbstractMatrix{T2}; disc=false, her=false) where {T1,T2}
+   T = promote_type(T1, T2)
+   GeneralizedLyapunovMap(convert.(AbstractMatrix{T}, (A, E))...; disc=disc, her=her)
 end
-lyapop(F::GeneralizedSchur{T,Matrix{T}}; kwargs...) where T = GeneralizedLyapunovMap(F.S, F.T; kwargs...)
+lyapop(F::GeneralizedSchur; kwargs...) = GeneralizedLyapunovMap(F.S, F.T; kwargs...)
 function lyapop(A::Number, E::Number; kwargs...)
-    T = promote_type(eltype(A),eltype(E))
-    GeneralizedLyapunovMap(A*ones(T,1,1), E*ones(T,1,1); kwargs...)
-end
-function lyapop(A::AbstractMatrix{T1}, E::AbstractMatrix{T2}; kwargs...) where {T1,T2}
-   T = promote_type(eltype(A),eltype(E)) 
-   GeneralizedLyapunovMap(convert(Matrix{T},A), convert(Matrix{T},E); kwargs...)
+   A, E = promote(A, E)
+   GeneralizedLyapunovMap(fill(A, 1, 1), fill(E, 1, 1); kwargs...)
 end
 function Base.size(L::GeneralizedLyapunovMap)
-     n = size(L.A,1)
-     N = L.her ? Int(n*(n+1)/2) : n*n
-     return (N,N)
+   n = size(L.A, 1)
+   N = L.her ? Int(n * (n + 1) / 2) : n * n
+   return (N, N)
 end
-Base.size(L::Adjoint{<:Any, <: GeneralizedLyapunovMap}) = size(L.parent)
-function LinearAlgebra.mul!(y::AbstractVector, L::GeneralizedLyapunovMap{T}, x::AbstractVector) where T
-   n = size(L.A,1)
+function mul!(y::AbstractVector, L::GeneralizedLyapunovMap{T}, x::AbstractVector) where T
+   n = size(L.A, 1)
    T1 = promote_type(T, eltype(x))
    if L.her
-      T == T1 ?  X = vec2triu(x, her = true) :  X = vec2triu(convert(Vector{T1}, x), her = true)
+      X = vec2triu(T == T1 ? x : convert(Vector{T1}, x), her=true)
```

Comment:
I wonder why you do all the conversion stuff manually. What would happen if `X` was real and `L` complex?